### ISSUE TITLE
Question about sample concatenation during training

### ISSUE THREAD

**Issue by mrzjy** [THUDM/GLM-130B#30]
Hi,

Thanks for your work and open-source!

There's one point I'm confused about: In your paper (section 2.3, last paragraph), you said

> For the [MASK] and multi-task objectives, we use a context window of 512 and concatenate four samples together to cater the 2,048-sequence-length

I wonder if there's a special attention mask to ensure that each sample should only attend to itself, and not attend to other samples? 
(e.g., something like a block diagonal attention mask as the following, where each block corresponds to one sample, respectively?)
![screenshot](https://user-images.githubusercontent.com/23355923/196839927-6f77ca05-2e84-4bc2-83e6-e7b700f96571.jpg)

Otherwise it would be weird to concatenate multiple independent samples together just for computation efficiency, or am I missing something here? (Since there's no training code in the repo yet)

**Comment by Sengxian** [comment]
Sure, we make the block diagonal attention mask matrix to ensure each sample only attend to itself. and construct the position_id like [0, 1, ..., 511, 0, 1, ..., 511, 0, 1, ..., 511, 0, 1, ..., 511] to make the result the same as they are individual calculated.

**Comment by mrzjy** [comment]
Thanks for your clear response!
